src/services/loggerInterface.ts

```diff
@@ -0,0 +1,48 @@
+export interface ILogger {
+    emerg(message: string): void;
+    alert(message: string): void;
+    crit(message: string): void;
+    error(message: string): void;
+    warning(message: string): void;
+    notice(message: string): void;
+    info(message: string): void;
+    debug(message: string): void;
+}
+
+export abstract class Logger implements ILogger {
```

Comment:
It would be better to put the abstract class in its own file because there will be cases when we will need the interface only